### ISSUE TITLE
Benchmark running the EMNIST and Shakespeare

### ISSUE THREAD

**Issue by luke-avionics** [FedML-AI/FedML#63]
Hi, is there any example settings to run EMNIST and Shakespeare? It seems to be very hard to get the results(that accuracy) in Adaptive Federated Optimization https://arxiv.org/pdf/2003.00295.pdf.

Thanks!

**Comment by luke-avionics** [comment]
Got it for emnist.

In model definition, relu should be added for each covn2d and linear layer (as seen in their released code ,.... not their paper)

The final softmax should be removed, as CrossEntropy in pytorch already include it. Or potentially could be replaced with LogSoftmax

Gradient clipping should also be added during training (probably should not be added to the code but ask the user to add it instead when running emnist)..... otherwise with paper reported learning rate it test acc will collapse at some point.

**Comment by chaoyanghe** [comment]
@luke-avionics Thanks for your comment. Let's chat at Slack.

**Comment by chaoyanghe** [comment]
@luke-avionics please join our slack using the link at https://fedml.ai

**Comment by chaoyanghe** [comment]
@luke-avionics do you mean cnn.py? I just updated the code.
